packages/realm-server/tests/realm-server-test.ts

```diff
@@ -4248,6 +4250,101 @@ module('Realm Server', function (hooks) {
         '/_user response is correct',
       );
     });
+
+    test('sends billing notification on successful Stripe webhook handling', async function (assert) {
+      let matrixClient = new MatrixClient({
+        matrixURL: realmServerTestMatrix.url,
+        username: 'test_realm',
+        seed: secretSeed,
+      });
+      await matrixClient.login();
+      let userId = matrixClient.getUserId();
+
+      let authResponse = await request
+        .post('/_server-session')
+        .send(JSON.stringify({ user: userId }))
+        .set('Accept', 'application/json')
+        .set('Content-Type', 'application/json');
+
+      let json = authResponse.body;
+      let { joined_rooms: rooms } = await matrixClient.getJoinedRooms();
+      if (!rooms.includes(json.room)) {
+        await matrixClient.joinRoom(json.room);
+      }
+
+      await matrixClient.sendEvent(json.room, 'm.room.message', {
+        body: `auth-response: ${json.challenge}`,
+        msgtype: 'm.text',
+      });
+
+      await request
+        .post('/_server-session')
+        .send(JSON.stringify({ user: userId, challenge: json.challenge }))
+        .set('Accept', 'application/json')
+        .set('Content-Type', 'application/json');
+
+      const secret = process.env.STRIPE_WEBHOOK_SECRET;
+      await insertUser(dbAdapter, userId!, 'cus_123');
+      await insertPlan(dbAdapter, 'Free plan', 0, 100, 'prod_free');
+      if (!secret) {
+        throw new Error('STRIPE_WEBHOOK_SECRET is not set');
+      }
+      let event = {
```

Comment:
Here you are testing the `invoice.payment_succeeded` webhook only, while we are handling several other handlers too. 

To avoid repeating other events here I am wondering if there should (also) be a general test that makes sure `stripeWebhookHandler` will notify the user after any of its available handlers are called.

Reply:
Since we already have a test for payment deleted event, I added the assertion there for billing notification and I also created a test for checkout session completed event.